### DISCUSSION TITLE
how to fix  error in Tutorial 5 Multi Assets Algorithmic Trading Backtesting with Backtrader

### DISCUSSION THREAD

**Discussion by smileandsaydl** [dcajasn/Riskfolio-Lib#42] [General]
Hi dcajasn
    I am newcomer，when I run the follow code , I get error " The problem doesn't have a solution with actual input parameters" 
   What may have been the reason that the ‘solution’ couldn’t be found? Where should I look at to find the fix? Thanks,
 
follow code   
rms = ['MV', 'CVaR', 'WR', 'CDaR']

for j in rms:
    
    weights = pd.DataFrame([])

    for i in index_:
        Y = returns.iloc[i-1000:i,:] # taking last 4 years (250 trading days per year)

        # Building the portfolio object
        port = pf.Portfolio(returns=Y)  
        # Add portfolio constraints
        port.ainequality = A
        port.binequality = B
        
        # Calculating optimum portfolio

        # Select method and estimate input parameters:

        method_mu='hist' # Method to estimate expected returns based on historical data.
        method_cov='hist' # Method to estimate covariance matrix based on historical data.

        port.assets_stats(method_mu=method_mu, method_cov=method_cov, d=0.94)
        
        # Estimate optimal portfolio:
        
        port.solvers = ['MOSEK']
        port.alpha = 0.05
        model='Classic' # Could be Classic (historical), BL (Black Litterman) or FM (Factor Model)
        rm = j # Risk measure used, this time will be variance
        obj = 'Sharpe' # Objective function, could be MinRisk, MaxRet, Utility or Sharpe
        hist = True # Use historical scenarios for risk measures that depend on scenarios
        rf = 0 # Risk free rate
        l = 0 # Risk aversion factor, only useful when obj is 'Utility'

        w = port.optimization(model=model, rm=rm, obj=obj, rf=rf, l=l, hist=hist)

        if w is None:
            w = weights.tail(1).T
        weights = pd.concat([weights, w.T], axis = 0)
    
    models[j] = weights.copy()
    models[j].index = index_

**Comment by dcajasn** [comment]
The message means that the problem doesn't have a solution. The two most common reasons for this message are:
- The problem doesn't have a solution for your input data. Portfolio object uses convex optimization models to find optimal weights, so some values and constraints can make the problem unsolvable.
- The solver doesn't support the model you are trying to solve. In this case you must check CVXPY's docs and check the paper related to the model you are trying to solve.
